### DISCUSSION TITLE
Não consegui instalar no mac

### DISCUSSION THREAD

**Discussion by porres** [charlesneimog/py4pd#65] [General]
oi, to usando mac intel 10.14.6, installei o python 3.11 antes mas nao rolou usar o external

![Screen Shot 2023-11-23 at 13 31 59](https://github.com/charlesneimog/py4pd/assets/5360142/cad84d66-6d61-455d-968e-34ff0060b8e6)

**Comment by charlesneimog** [comment]
Feliz que vc ta testando esse objeto rsrsrs. 

Eu vou dar uma olhada e instalar um máquina virtual e ver. Esse objeto vai ter os mesmo problemas que o sfont~ eu acho.

**Comment by porres** [comment]
to testando de alegre, sei nada de python, só de monty python

**Comment by charlesneimog** [comment]
Ahhh, mas é legal rsrs, Pd vai ficar o OM com esteroides

**Comment by porres** [comment]
nao tenho a mínima ideia como... vai dar pra visualizar partitura e tudo? 

O louco e ideal seria, como no sfont~, ter o python como uma lib dinâmica, rs

**Comment by charlesneimog** [comment]
A ideia é, mas preciso resolver esses negocios primeiro. O ruim é que não rola distribuir o Python como uma lib dynamica. Eu tentei ....


Da uma olhada

https://github.com/charlesneimog/py4pd/blob/eebf0325512d8a0113520c5ab269cd971f2294ca/docs/examples/score/score.gif

**Comment by porres** [comment]
agora fudeu :)

mas vc precisa do python pra desenhar a parte gráfica?

**Comment by charlesneimog** [comment]
Daria para programar em C, mas é muito mais fácil fazer em Python, outra coisa que me interesso são as bibliotecas. Tipo essa neoscore do Python é muito poderosa, até que escrever tudo que ela tem em C levaria anos eu acho AHAHHA. Além das dores de cabeça com o Tk.

Edit: Daria, mas eu não conseguiria n... Ahah

**Comment by porres** [comment]
Tendi, mas em tese daria pra fazer em tk né 

Putz, acho que preciso disso na else , e agora? Kkkk

**Comment by porres** [comment]
Mas eu não entendo como o Python desenha coisas gráficas no Pd, explica?

**Comment by porres** [comment]
Uma boa ideia seria colocar isso no plugdata hehe

**Comment by charlesneimog** [comment]
Na real, debaixo dos panos eu estou usando uma imagem, o código que faz isso e bastante coisa copiada do else/pic. Eu acho, que se vc achar uma lib em C que consegue gerar as partituras, vc consegue mostrar usando o pic... 

Eu já dei uma olhada no plugdata, mas e meio complicado, pelo menos n vai dar antes deles implementarem alguma forma de saber em qual Pd eu estou rodando.

**Comment by charlesneimog** [comment]
e muito mais simples do que parece ahahha

**Comment by charlesneimog** [comment]
O legal cm isso e que vc pode mostrar outras coisas, sonogramas, gráficos etc... 

Por exemplo, cm o python conseguimos usar vamp plugins, daí vc tem tipo um Sonic visualizer dentro do PD.


Enfim, eu acho bem legal ahahay

**Comment by porres** [comment]
> e muito mais simples do que parece ahahha

 o q?

**Comment by porres** [comment]
eu piro na gente fazer um "bach" pro Pd

**Comment by charlesneimog** [comment]
Mas esse objeto é isso mesmo. Mas em vez de usar aquelas libs llll, saca? Eu uso um negócio chamado PyObject.

**Comment by charlesneimog** [comment]
> > e muito mais simples do que parece ahahha
> 
>  o q?

De mostrar a partitura...

**Comment by porres** [comment]
> Mas esse objeto é isso mesmo. Mas em vez de usar aquelas libs llll, saca? Eu uso um negócio chamado PyObject.

não entendi nada

tá em sp ainda, quer colar amanhã na casa? (muito próprio isso aqui no github, hahahah)
